Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -4502,7 +4502,10 @@ bool RenderLayer::hitTest(const HitTestRequest& request, const HitTestLocation&
         // We didn't hit any layer. If we are the root layer and the mouse is -- or just was -- down,
         // return ourselves. We do this so mouse events continue getting delivered after a drag has 
         // exited the WebView, and so hit testing over a scrollbar hits the content document.
-        if (!request.isChildFrameHitTest() && (request.active() || request.release()) && isRenderViewLayer()) {
+        // In addtion, it is possible for the mouse to stay in the document but there is no element.
+        // At that time, the events of the mouse should be fired.
+        LayoutPoint hitPoint = hitTestLocation.point();
+        if (!request.isChildFrameHitTest() && ((request.active() || request.release()) || (request.move() && hitTestArea.contains(hitPoint.x(), hitPoint.y()))) && isRenderViewLayer()) {
```

Comment:
This condition is getting long, do you mind breaking out the new logic into a separate predicate? Also, `LayoutRect::contains(const LayoutPoint&)` is a thing, so your `contains()` call can be made more concise.

```suggestion
        bool moveRequestIsOverDocument = request.move() && hitTestArea.contains(hitPoint);
        if (!request.isChildFrameHitTest() && (request.active() || request.release() || moveRequestIsOverDocument) && isRenderViewLayer()) {
```

Reply:
Done.